api_v3/lib/types/entry/KalturaLiveEntry.php

```diff
@@ -208,15 +208,12 @@ public function doFromObject($dbObject, KalturaDetachedResponseProfile $response
 			$this->recordingOptions->fromObject($dbObject->getRecordingOptions());
 		}
 
-		if ($dbObject->getExplicitLive())
+		if ($dbObject->getViewMode() == ViewMode::ALLOW_ALL && in_array($dbObject->getLiveStatus(), array(KalturaEntryServerNodeStatus::PLAYABLE, KalturaEntryServerNodeStatus::BROADCASTING)))
```

Comment:
For better reading please move to small private function - shouldSetRedirectEntryId()
Or just getRedirectEntryId() and it can manage the logic